src/App.jsx

```diff
@@ -1,32 +1,24 @@
-import React, { useEffect } from 'react';
+import React from 'react';
 
-import { useDispatch } from 'react-redux';
+import { Switch, Route, Link } from 'react-router-dom';
 
-import RegionsContainer from './RegionsContainer';
-import CategoriesContainer from './CategoriesContainer';
-import RestaurantsContainer from './RestaurantsContainer';
-
-import {
-  loadInitialData,
-} from './actions';
-
-// 0. 지역, 분류 목록을 얻기
-// 1. 지역 선택 - Regions <- API (0)
-// 2. 분류 선택 - Categories - 한식, 중식, 일식, ... <- API (0)
-// 3. 식당 목록 - Restaurants <- API (with region, category) -> 1, 2 모두 완료된 경우
+import HomePage from './HomePage';
+import AboutPage from './AboutPage';
+import RestaurantsPage from './RestaurantsPage';
+import RestaurantPage from './RestaurantPage';
+import NotFoundPage from './NotFoundPage';
 
 export default function App() {
-  const dispatch = useDispatch();
-
-  useEffect(() => {
-    dispatch(loadInitialData());
-  });
-
   return (
     <div>
-      <RegionsContainer />
-      <CategoriesContainer />
-      <RestaurantsContainer />
+      <h1><Link to="/">헤더</Link></h1>
+      <Switch>
+        <Route exact path="/" component={HomePage} />
+        <Route path="/about" component={AboutPage} />
+        <Route exact path="/restaurants" component={RestaurantsPage} />
+        <Route path="/restaurants/:restaurantId" component={RestaurantPage} />
```

Comment:
```suggestion
        <Route path="/restaurants/:id" component={RestaurantPage} />
```

보통 REST에서 URL을 표현할 땐 id로 심플하게 얘기합니다. 무언가 식별자(id)로 구분한다는 게 중요합니다.
그게 이름인지 뭔지는 모르지만 각 레스토랑을 구분하는 식별자(id)가 있다는 거죠.